host/xtest/install_ta.c

```diff
@@ -45,15 +45,20 @@ static void *read_ta(const char *dname, const char *fname, size_t *size)
 		err(1, "fseek");
 
 	s = ftell(f);
+	if (s < 0) {
+		fclose(f);
+		return NULL;
+	}
 	rewind(f);
 
 	buf = malloc(s);
 	if (!buf)
 		err(1, "malloc(%zu)", s);
 
-	if (fread(buf, 1, s, f) != s)
+	if (fread(buf, 1, s, f) != (size_t)s)
 		err(1, "fread");
 
+	fclose(f);
 	*size = s;
```

Comment:
should cast `s` here also.

That make many casts.
maybe prefer to add another local var: `long l; (...) l = ftell(f); if (l < 0) err(...); s = (size_t)l; (...)`

Reply:
I don't understand what your comment has to do with this commit.  Are you commenting on another fix that could be done to the code?

Reply:
@etienne-lms is suggesting the same thing I was about to suggest. Add another helper variable to store the return value from `ftell()` and keep the type of `s` as is.